src/Tests/Shared/PlatformConditionAttribute.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+[Flags]
+public enum Platform
+{
+    Windows32 = 1 << 0,
+    Windows64 = 1 << 1,
+    Windows = Windows32 | Windows64,
+    Linux32 = 1 << 2,
+    Linux64 = 1 << 3,
+    Linux = Linux32 | Linux64,
+    OSX32 = 1 << 4,
+    OSX64 = 1 << 5,
+    OSX = OSX32 | OSX64,
+    Unix32 = Linux32 | OSX32,
+    Unix64 = Linux64 | OSX64,
+    Unix = Unix32 | Unix64,
+}
+
+/// <summary>
+///     This attribute is used to execute a test class or a test
+///     method only if the platform matches, with an optional message.
+/// </summary>
+[AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, Inherited = false)]
+public sealed class PlatformConditionAttribute : ConditionBaseAttribute
+{
+    private readonly Platform _platform;
+
+    /// <summary>
+    ///     Initializes a new instance of the <see cref="PlatformConditionAttribute" /> class.
+    /// </summary>
+    /// <param name="mode">An enumeration used with to control whether the condition is reversed.</param>
+    /// <param name="platform">The platform that this test supports.</param>
+    public PlatformConditionAttribute(ConditionMode mode, Platform platform)
+        : base(mode)
+    {
+        _platform = platform;
+        IgnoreMessage = $"Test is supported only on {platform}";
+    }
+
+    /// <summary>
+    ///     Initializes a new instance of the <see cref="PlatformConditionAttribute" /> class.
+    /// </summary>
+    /// <param name="platform">The platform that this test supports.</param>
+    public PlatformConditionAttribute(Platform platform)
+        : this(ConditionMode.Include, platform)
+    {
+    }
+
+    public override bool ShouldRun
+    {
+        get
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                if (Environment.Is64BitOperatingSystem)
+                {
+                    return (_platform & Platform.Windows64) != 0;
+                }
+
+                return (_platform & Platform.Windows32) != 0;
+            }
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                if (Environment.Is64BitOperatingSystem)
+                {
+                    return (_platform & Platform.Linux64) != 0;
+                }
+
+                return (_platform & Platform.Linux32) != 0;
+            }
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                if (Environment.Is64BitOperatingSystem)
+                {
+                    return (_platform & Platform.OSX64) != 0;
+                }
+
+                return (_platform & Platform.OSX32) != 0;
+            }
+
+            return false;
+        }
+    }
+
+    public override string? IgnoreMessage { get; }
```

Comment:
I think it is possible to define the ignore message as not nullable but I'm not 100% sure. Can you try?